llvm-external-projects/iree-dialects/include/iree-dialects/Transforms/TransformMatchers.h

```diff
@@ -40,9 +40,10 @@ struct AllDims {};
 /// for all operands of the relevant kind.
 struct AllOperands {};
 
-struct CaptureDim {
-  explicit CaptureDim(int64_t &value) : value(value) {}
-  int64_t &value;
+template <typename T>
+struct CaptureStaticValue {
+  explicit CaptureStaticValue(T &value) : value(value) {}
+  T &value;
```

Comment:
This is okay, but we should preserve the named classes for matcher readability. They can derive this class.